### ISSUE TITLE
Sensitive/Deferred not working for podman::image

### ISSUE THREAD

**Issue by tuxmaster5000** [southalc/podman#32]
When the image will live on an private then, the login credentials must be given to podman::image.
In the theory it will work with the flags settings, but in the reality it will fail.
Sample code:
```
$user = Deferred(FUNCTION,[OPTION])
 $pw = Deferred(FUNCTION,[OPTION])
$cred = {
  creds => Sensitive(Deferred(sprintf, ['%s:%s', $user, $pw]))
}
podman::image { 'foo':
  ensure => present,
  image => 'REGISTRY/REPO/IMAGE:TAG',
  flags   => $cred
}
```
it will fails with:
rejected: parameter 'enumerable' expects an Iterable value, got Sensitive[Deferred] modules/podman/manifests/image.pp, line: 48, column: 29)

**Comment by southalc** [comment]
I think you might have to use unwrap() in order to reference the sensitive value.  See here and let me know if this works for you:
https://puppet.com/docs/puppet/7/lang_data_sensitive.html

**Comment by tuxmaster5000** [comment]
Hm, when I call flags   => $cred.unwrap, it will result in the same.

**Comment by southalc** [comment]
This will be addressed if/when issue #23 is implemented to support podman login.
